src/logger/logger.py

```diff
@@ -78,26 +79,94 @@ def set_signal(
         """
         pass  # not implemented yet # pylint: disable=W0107
 
-    def inject_fault(
-        self, injection_id: int, fault_type: int, injection_position: any, duration: int
+    def inject_train_speed_fault(
+        self,
+        train_speed_fault_configuration: UUID,
+        affected_element: str,
+        value_before: str,
+        value_after: str,
     ) -> Type[None]:
         """
-        This function should be called when injecting a fault into the simulation. This should
-        include (the injection id), type of injection, position of injection, duration and some
-        information if required. (Maybe adapt later)
-        :param injection_id: The id of the injection
-        :param fault_type: The fault type
-        :param injection_position: The position of the injection
-        :param duration: The duration of the injection
+        This function should be called when injecting a train speed fault into the simulation.
+        This should include the fault configuration, the affected element, the value before and the
+        value after the fault.
+        :param train_speed_fault_configuration: The configuration of the fault
+        :param affected_element: The affected element
+        :param value_before: The value before the fault
+        :param value_after: The value after the fault
         :rtype: None
         """
         pass  # not implemented yet # pylint: disable=W0107
 
-    def resolve_fault(self, injection_id: int) -> Type[None]:
+    def inject_platform_blocked_fault(
+        self,
+        platform_blocked_fault_configuration: UUID,
+        affected_element: str,
+        value_before: str,
+        value_after: str,
+    ) -> Type[None]:
+        """
+        This function should be called when injecting a platform blocked fault into the simulation.
+        This should include the fault configuration, the affected element, the value before and the
+        value after the fault.
+        :param platform_blocked_fault_configuration: The configuration of the fault
+        :param affected_element: The affected element
+        :param value_before: The value before the fault
+        :param value_after: The value after the fault
+        :rtype: None
+        """
+
+    pass  # not implemented yet # pylint: disable=W0107
+
+    def inject_cancelled_speed_fault(
```

Comment:
missing `train` but this is irrelevant due to the renaming of `TrainCancelledFault` to `ScheduleBlockedFault`

Reply:
added it for completeness